src/deploy/scope_managementgroup.ts

```diff
@@ -67,18 +66,23 @@ export async function DeployManagementGroupScope(azPath: string, region: string,
         // execute the deployment
         core.info("Creating deployment...")
         var deploymentCode = await exec(`"${azPath}" deployment mg create ${azDeployParameters} -o json`, [], deployOptions);
-        if (commandStdErr.trim().length !== 0) {
-            throw new Error(`Deployment process failed as some lines were written to stderr: ${commandStdErr}`)
-        } else {
-            if (deploymentCode != 0) {
-                core.error("Deployment failed.")
+        
+        if (deploymentCode == 0 && commandStdErr.trim().length !== 0) {
```

Comment:
can this be simplified by:

First log the error wherever they get into stderr listener and set a flag as well

```
stderr: (data: BufferSource) => {
                error = data.toString();
                if(error && error.trim().length !== 0)
                {
                    commandStdErr = true;
                    core.error(error);
                }
            }
```


```
        if (deploymentCode == 0) {
            throw new Error("Deployment failed.")
        }
        if(commandStdErr && failOnStdErr) {
            throw new Error("Deployment process failed as some lines were written to stderr");
        }
        
        core.debug(commandOutput);
        core.info("Parsing outputs...")
        return ParseOutputs(commandOutput)
```